sdk/storage/azblob/container/client.go

```diff
@@ -329,3 +333,40 @@ func (c *Client) GetSASURL(permissions sas.ContainerPermissions, expiry time.Tim
 
 	return endpoint, nil
 }
+
+// NewBatchBuilder creates an instance of BatchBuilder using the same auth policy as the client.
+// BatchBuilder is used to build the batch consisting of delete or set tier sub-requests or both.
+func (c *Client) NewBatchBuilder() (*BatchBuilder, error) {
+	conOptions := new(ClientOptions)
+	if c.authPolicy() != nil {
+		conOptions.PerRetryPolicies = append(conOptions.PerRetryPolicies, c.authPolicy())
+	}
+
+	// TODO: Use an empty transport so requests aren't sent
+	conOptions.Transport = nil
+
+	pl := runtime.NewPipeline(exported.ModuleName, exported.ModuleVersion, runtime.PipelineOptions{}, &conOptions.ClientOptions)
```

Comment:
I suspect that you don't really need to create a full pipeline as it contains a bunch of policies that aren't useful for this.  You should just be able to call `Policy.Do()` in order to add the necessary auth content to the fake request.

Reply:
Do you mean we should not create a new pipeline for the BatchBuilder and directly use `Policy.Do()`. If this is the case, we don't need `pipeline` field in the BatchBuilder. It will look like this,
```
type BatchBuilder struct {
	endpoint    string
	subRequests []*policy.Request
}
```

If we do want to have a separate pipeline for the BatchBuilder, can you suggest some way to create a pipeline having just the auth policy (we can add a dummy transport policy to it if required).

Reply:
There is no way to create an empty pipeline.  We _could_ add this if necessary but I'm not convinced we need it.